frontend/src/components/bucket/BucketConfigForm.vue

```diff
@@ -77,14 +77,35 @@ const onSubmit = async (values: any) => {
       formBucket.key = joinPath(values.key);
     }
 
+    const bucketChanges = differential(formBucket, initialValues);
+
     props.bucket
-      ? await bucketStore.updateBucket(props.bucket?.bucketId, differential(formBucket, initialValues))
+      ? await bucketStore.updateBucket(props.bucket?.bucketId, bucketChanges)
       : await bucketStore.createBucket(formBucket);
 
     await bucketStore.fetchBuckets({ userId: getUserId.value, objectPerms: true });
+
+    // trim trailing "//", if present
+    // (modifying getBucketPath() instead seems to break nested bucket display [PR-146])
+    const currBucketPath = getBucketPath(initialValues as Bucket).endsWith('//')
+      ? getBucketPath(initialValues as Bucket).slice(0, -1)
```

Comment:
dont you need to call `getBucketPath(formBucket as Bucket)` using the `formBucket` (updated properties) rather than initialValues?

Reply:
`getBucketPath()` expects the bucket key, which `formBucket` doesn't have since you aren't really (nor can you) change the path for an existing COMS bucket. If the bucket doesn't point at a subdirectory in the bucket you'll end up with `undefined` tacked onto the end where it expects the bucket sub-path (e.g. `https://s3.amazonaws.com/yxwgj/undefined`).

That helper function was lifted as-is from `BucketTable` into `utils.ts`, so I'm a bit wary of changing how it works to accommodate this.

Reply:
sorry. i forgot you can't edit the bucket key. initialValues is fine